Gems/EditorPythonBindings/Code/Source/ActionManager/PythonEditorAction.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/std/function/function_template.h>
+
+#include <Source/PythonCommon.h>
+#include <pybind11/pybind11.h>
+
+class PythonEditorAction
+{
+public:
+    AZ_TYPE_INFO(PythonEditorAction, "{1A5676D2-767B-4C2F-BC35-9CDDCE1430BB}");
+    AZ_CLASS_ALLOCATOR(PythonEditorAction, AZ::SystemAllocator, 0);
+
+    explicit PythonEditorAction(PyObject* handler);
+    PythonEditorAction(const PythonEditorAction& obj);
+    virtual ~PythonEditorAction();
+
+    PyObject* GetHandler() const;
```

Comment:
update: This should not be `const` if it's returning a non-const pointer (either remove the `const` or have the return be `const` (or have a non-const overload)

Reply:
This is actually a bit of a problematic function and this is the only way I could make it work for now:

```
    AzToolsFramework::ActionManagerOperationResult PythonEditorActionHandler::RegisterAction(
        const AZStd::string& contextIdentifier,
        const AZStd::string& identifier,
        const AZStd::string& name,
        const AZStd::string& description,
        const AZStd::string& category,
        const AZStd::string& iconPath,
        PythonEditorAction handler)
    {
        return m_actionManagerInterface->RegisterAction(
            contextIdentifier, identifier, name, description, category, iconPath,
            [h = AZStd::move(handler)]()
            {
                PyObject_CallObject(h.GetHandler(), NULL);
            });
    }
```

- We move the PythonEditorAction object to the lambda that gets registered to the Action Manager here:
This makes the h object const.
- We then pass the PyObject* to PyObject_CallObject, it needs to be non-const for the function to build.

Technically GetHandler() does not alter the state of the PythonEditorAction instance, although it gives out direct access to the PyObject embedded in it... I'd honestly prefer to either have a fully const or fully non-const function, but I'm not sure there's a way to do it with the current setup?

Reply:
You can provide both overloads? But would that then cause a compile error? If it's causing a problem it might hint at an issue with the design that could be slightly reworked (could always sync up offline to discuss :+1:)

Reply:
If I add both overloads, it would still error out since it would not be able to use either.

Reply:
We had a chat and found a working fix, thanks Tom!